src/SQLite3-Core-Tests/SQLiteConnectionTest.class.st

```diff
@@ -48,9 +49,39 @@ SQLiteConnectionTest >> setUp [
 SQLiteConnectionTest >> tearDown [
 	
 	[ db close ] on: SQLite3Misuse do: [ ].
+	[ target ifNotNil:[target close] ] on: SQLite3Misuse do: [ ].
 	super tearDown.
 ]
 
+{ #category : #tests }
+SQLiteConnectionTest >> testBackup [
+	| row backup |
+	
+	target := SQLite3Connection memory.
+	target open.
+	self populateRowsInto: 'junk' of: db.
+	backup := db backupTo: target.
+	backup open; step; finish.
+	row := (target execute: 'select * from junk where rt = ''1-ZZZ''') onlyRow.
```

Comment:
This is testing the target not the backup. No?

Reply:
The backup is a backup object - it is a process kind of object like a statement.  It has step's that you can repeatedly call to push the backup process forwards.  You could implement a progress indicator with it.

The target is the database we are copying db to.  If target has the same data as db, then the backup succeeded.

Reply:
sqlite3_backup* sqlite3_backup_init(
                    sqlite3* db_dst, const char* db_name_dst,
                    sqlite3* db_src, const char* db_name_src );

backup is the pharo object representing sqlite3_backup.

Check out https://www.oreilly.com/library/view/using-sqlite/9781449394592/re225.html